src/pages/system-checks/index.js

```diff
@@ -0,0 +1,121 @@
+import { useState, useContext, useEffect } from 'react'
+import { Box } from '@mui/material'
+import toast from 'react-hot-toast'
+import Table from 'src/components/Shared/Table'
+import WindowToolbar from 'src/components/Shared/WindowToolbar'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { useWindowDimensions } from 'src/lib/useWindowDimensions'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { DataSets } from 'src/resources/DataSets'
+import { CommonContext } from 'src/providers/CommonContext'
+import useResourceParams from 'src/hooks/useResourceParams'
+
+const SystemChecks = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { height } = useWindowDimensions()
+  const { getAllKvsByDataset } = useContext(CommonContext)
+  const [data, setData] = useState([])
+
+  const { labels: _labels, access } = useResourceParams({
+    datasetId: ResourceIds.SystemChecks
+  })
+
+  const handleSubmit = () => {
+    postChecks()
+  }
+
+  async function getAllSystems() {
+    return new Promise((resolve, reject) => {
+      getAllKvsByDataset({
+        _dataset: DataSets.SYSTEM_CHECKS,
+        callback: result => {
+          if (result) resolve(result)
+          else reject()
+        }
+      })
+    })
+  }
+
+  useEffect(() => {
+    ;(async function () {
+      const systemCheckData = await getAllSystems()
+
+      const resCheckedSystems = getRequest({
+        extension: SystemRepository.SystemChecks.qry,
+        parameters: `_scope=1`
+      })
+
+      Promise.all([resCheckedSystems]).then(([checkedSystems]) => {
```

Comment:
Why do we have Promise.all if we have only one request? Can this be replaced with useResourceQuery?